### ISSUE TITLE
Slow require/import on serverless functions

### ISSUE THREAD

**Issue by mark-veenstra** [twilio/twilio-node#525]
### Issue Summary
Requiring the `twilio-node` library takes a few seconds when using it in a Docker container or in an own server setup, maybe the issue is not that important. Since it will load into memory and then it is available during the lifetime of the Node process.

But if you want to use `twilio-node` in a serverless environment cold starts of the function can be done very often. It seems that all functions where we require/import `twilio-node` takes some time.

Is there anyway to import parts of the `twilio-node` package instead of setting everything up? I think that may speed up the import/require. Or any other ideas?

### Steps to Reproduce
1. Create a serverless function on Firebase or AWS
2. Add code like:
```node
import * as twilio       from 'twilio';
const client = twilio(process.env.TWILIO_ACCOUNT_SID, process.env.TWILIO_AUTH_TOKEN);
console.log('Loaded');
```

### Exception/Log
On Firebase Cloud Function the execution takes:
```
Function execution took 2872 ms, finished with status code: 200
```

### Technical details:
* twilio-node version: 3.39.2
* node version: 8.10.1

**Comment by nahuef** [comment]
Probably not related, but AWS dropped support for node 8.x, you should update your runtime.
[More info](https://acloud.guru/forums/aws-lambda/discussion/-LrVfksaEiXxdS1pRMKf/Node.js%208.10%20has%20reached%20its%20End%20Of%20Life)

**Comment by mark-veenstra** [comment]
On Firebase default is still node 8, but even when we use node 10 the problem exists. Partial initialization of the `twilio-node` would help.

**Comment by wolfenrain** [comment]
I have done some tweaking and added a proof of concept lazy loading to the Twilio module to see if it improves the loading time, see the [compare](https://github.com/twilio/twilio-node/compare/3d0e4a1de19476337f2ac4413b6c612c58b3c721...Mediapioniers:faster-requires) for reference.

When lazy loading is disabled(default), test returns a 600ms timeframe:
```javascript
console.time('Twilio');
const Twilio = require('./index');
const client = new Twilio('AAAAAAAAAAAAAAAAAAAAAAAA', 'AUTHTOKEN');
console.timeEnd('Twilio');
```

When lazy loading is enabled, test returns a 16ms timeframe:
```javascript
console.time('Twilio');
const Twilio = require('./index');
const client = new Twilio('AAAAAAAAAAAAAAAAAAAAAAAA', 'AUTHTOKEN', { lazyLoading: true });
console.timeEnd('Twilio');
```

The require of Twilio has been changed slightly for the `initializer`, instead of auto requiring everything, I made it into getter. This already gave a small increase. But the biggest change was removing all the `require`s from `Twilio.js`. I moved those to getters as well.

And the requiring of the `RestClient.js` is also a big time consumer, so I made a getter for that as well.

Problem is that the `Twilio.js` file is automatically generated.. but my compare should at least show that certain optimizations are possible. It would be great if the code generator is able to generate these optimizations.
